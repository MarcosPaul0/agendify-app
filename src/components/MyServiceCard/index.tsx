import { Text, TouchableOpacity, View } from 'react-native';
import { Image } from 'expo-image';
import { Tag } from '@components/Tag';
import { useRouter } from 'expo-router';
import { APP_ROUTES } from '@constants/appRoutes.constant';
import { BASE_URL } from '@constants/baseUrl.constant';
import { IMyServiceCardProps } from './interfaces/myServiceCardProps.interface';

export function MyServiceCard({ service, isLastItem }: IMyServiceCardProps) {
  const router = useRouter();

  function navigateToManageService() {
    router.push(`${APP_ROUTES.MANAGE_SERVICE}/${service.id}`);
  }

  return (
    <TouchableOpacity
      onPress={navigateToManageService}
      className={`
        rounded-lg bg-GRAY_100 border border-GRAY_500 w-40
        py-2 px-4 items-center ml-2 h-40 ${isLastItem ? 'mr-5' : ''}
      `}
    >
      <Image
        source={`${BASE_URL}/${service.imageUrl}`}
        className="w-full h-20 border rounded-lg border-BLUE_500"
      />

      <View className="w-full">
        <View className="w-full border-b border-b-GRAY_500 my-2">
          <Text className="font-bold text-BLUE_900 text-md text-center">
            {service.name}
          </Text>
        </View>

        <View className="flex-row items-center">
          {service.price && (
            <Tag
              text={`R$ ${String(service.price.toFixed(2)).replace('.', ',')}`}
              marginLeftIsActive={false}
              size="sm"
            />
          )}
          <Tag text={`${service.duration} h`} size="sm" />
        </View>
      </View>
    </TouchableOpacity>
  );
}
