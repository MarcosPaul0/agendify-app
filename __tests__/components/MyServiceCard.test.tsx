import { render } from '@testing-library/react-native';
import { MyServiceCard } from '@components/MyServiceCard';
import { service } from '../../mocks/service.mock';

jest.mock('expo-router', () => {
  const currentRouter = jest.requireActual('expo-router');

  return {
    ...currentRouter,
    useRouter: jest.fn().mockReturnValue({
      push: jest.fn().mockImplementation((path: string) => path),
    }),
  };
});

describe('My Service Card', () => {
  it('Match with snapshot', async () => {
    const tree = render(<MyServiceCard service={service} isLastItem />);

    expect(tree).toMatchSnapshot();
  });
});
