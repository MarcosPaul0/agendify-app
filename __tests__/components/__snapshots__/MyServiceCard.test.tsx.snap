// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Service Card Match with snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#F5F5F5",
      "borderBottomColor": "#ADB5BD",
      "borderBottomLeftRadius": 8,
      "borderBottomRightRadius": 8,
      "borderBottomWidth": 1,
      "borderLeftColor": "#ADB5BD",
      "borderLeftWidth": 1,
      "borderRightColor": "#ADB5BD",
      "borderRightWidth": 1,
      "borderTopColor": "#ADB5BD",
      "borderTopLeftRadius": 8,
      "borderTopRightRadius": 8,
      "borderTopWidth": 1,
      "height": 160,
      "marginLeft": 8,
      "marginRight": 20,
      "opacity": 1,
      "overflow": "hidden",
      "width": 160,
    }
  }
>
  <ViewManagerAdapter_ExpoImage
    height={80}
    onError={[Function]}
    onLoad={[Function]}
    onLoadStart={[Function]}
    onProgress={[Function]}
    proxiedProperties={
      {
        "contentFit": "cover",
        "contentPosition": {
          "left": "50%",
          "top": "50%",
        },
        "placeholder": [],
        "source": [
          {
            "uri": "http://10.0.2.2:3333/fake-service-image-url",
          },
        ],
        "tintColor": undefined,
        "transition": null,
      }
    }
    style={
      {
        "height": 80,
        "width": "100%",
      }
    }
    width="100%"
  />
  <View
    style={
      [
        {
          "width": "100%",
        },
        {
          "paddingLeft": 8,
          "paddingRight": 8,
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "width": "100%",
          },
          {
            "borderBottomWidth": 1,
          },
          {
            "borderBottomColor": "#ADB5BD",
          },
          {
            "marginTop": 4,
          },
          {
            "marginBottom": 8,
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "fontWeight": "700",
            },
            {
              "color": "#0D141F",
            },
            {
              "textAlign": "center",
            },
          ]
        }
      >
        fake-service-name
      </Text>
    </View>
    <View
      style={
        [
          {
            "flexDirection": "row",
          },
          {
            "alignItems": "center",
          },
        ]
      }
    >
      <Text
        numberOfLines={2}
        style={
          [
            {
              "fontSize": 12,
              "lineHeight": 16,
            },
            {
              "color": "#6C757D",
            },
          ]
        }
      >
        fake-service-description
      </Text>
    </View>
  </View>
</View>
`;
